src/condor_io/condor_secman.cpp

```diff
@@ -3292,26 +3274,19 @@ SecMan::ReconcileMethodLists( char * cli_methods, char * srv_methods ) {
 	// the output will be a list of methods supported by both, in the
 	// order that the server prefers.
 
-	StringList server_methods( srv_methods );
-	StringList client_methods( cli_methods );
-	const char *sm = NULL;
-	const char *cm = NULL;
-
 	std::string results;
 	int match = 0;
 
 	// server methods, one at a time
-	server_methods.rewind();
-	while ( (sm = server_methods.next()) ) {
-		client_methods.rewind();
-		if (!strcasecmp("TOKENS", sm) || !strcasecmp("IDTOKENS", sm) || !strcasecmp("IDTOKEN", sm)) {
+	for (auto& sm : StringTokenIterator(srv_methods)) {
+		if (!strcasecmp("TOKENS", sm.c_str()) || !strcasecmp("IDTOKENS", sm.c_str()) || !strcasecmp("IDTOKEN", sm.c_str())) {
 			sm = "TOKEN";
 		}
-		while ( (cm = client_methods.next()) ) {
-			if (!strcasecmp("TOKENS", cm) || !strcasecmp("IDTOKENS", cm) || !strcasecmp("IDTOKEN", cm)) {
+		for (auto cm: StringTokenIterator(cli_methods)) {
+			if (!strcasecmp("TOKENS", cm.c_str()) || !strcasecmp("IDTOKENS", cm.c_str()) || !strcasecmp("IDTOKEN", cm.c_str())) {
 				cm = "TOKEN";
```

Comment:
This isn't your change, but why are these `strcasecmp` and the above ones `strcmp` ?  If it doesn't matter, in the long run, I think I'd rather see them be strcmp, then change the literals to static std::strings which get operator== compared.

Reply:
Those strcmp()s are the exception for methods names in the security code. All of our docs use uppercase names, but most of the code accepts both cases. If we want to change to case-sensitive comparisons, I'd want to add code to change all externally-supplied values to uppercase.